kpops/api/__init__.py

```diff
@@ -120,6 +120,31 @@ def manifest_destroy(
         yield resource
 
 
+def manifest_reset(
+    pipeline_path: Path,
+    dotenv: list[Path] | None = None,
+    config: Path = Path(),
+    steps: set[str] | None = None,
+    filter_type: FilterType = FilterType.INCLUDE,
+    environment: str | None = None,
+    verbose: bool = True,
+    operation_mode: OperationMode = OperationMode.MANIFEST,
```

Comment:
what happens if the user would pass `OperationMode.MANAGED`? does that even make sense?

Reply:
Passing `OperationMode.MANIFEST` or `OperationMode.ARGO` would be the same. Passing the OperationMode MANAGED needs to make more sense for the manifest functions.